op-chain-ops/cmd/op-run-block/main.go

```diff
@@ -0,0 +1,364 @@
+	hdr, err := retry.Do[*types.Header](ctx, 10, retry.Exponential(), func() (*types.Header, error) {
+		r.logger.Info("fetching block header", "hash", hash)
+		return r.cl.HeaderByHash(ctx, hash)
+	})
+	if err != nil {
+		r.logger.Error("failed to get block header", "err", err, "hash", hash)
+		return nil
+	}
+	if hdr == nil {
+		r.logger.Warn("header not found", "hash", hash)
+	}
+	return hdr
+}
+
+func (r remoteChainCtx) GetTd(hash common.Hash, number uint64) *big.Int {
+	return big.NewInt(1)
+}
+
+func (r remoteChainCtx) Engine() consensus.Engine {
+	return r.consensusEng
+}
+
+func (r remoteChainCtx) GetHeader(hash common.Hash, u uint64) *types.Header {
+	if r.hdr.Hash() == hash {
+		return r.hdr
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	hdr, err := retry.Do[*types.Header](ctx, 10, retry.Exponential(), func() (*types.Header, error) {
+		r.logger.Info("fetching block header", "hash", hash, "num", u)
+		return r.cl.HeaderByNumber(ctx, new(big.Int).SetUint64(u))
+	})
+	if err != nil {
+		r.logger.Error("failed to get block header", "err", err, "hash", hash, "num", u)
+		return nil
+	}
+	if hdr == nil {
+		r.logger.Warn("header not found", "hash", hash, "num", u)
+	}
+	if got := hdr.Hash(); got != hash {
+		r.logger.Error("fetched incompatible header", "expectedHash", hash, "fetchedHash", got, "num", u)
+	}
+	return hdr
+}
+
+func Process(logger log.Logger, config *params.ChainConfig,
+	block *sources.RPCBlock,
+	statedb *gstate.StateDB, cfg vm.Config,
+	chainCtx *remoteChainCtx, outW io.Writer) (*core.ProcessResult, error) {
+	var (
+		receipts    types.Receipts
+		usedGas     = new(uint64)
+		header      = block.CreateGethHeader()
+		blockHash   = block.Hash
+		blockNumber = new(big.Int).SetUint64(uint64(block.Number))
+		allLogs     []*types.Log
+		gp          = new(core.GasPool).AddGas(uint64(block.GasLimit))
+	)
+
+	// Mutate the block and state according to any hard-fork specs
+	if config.DAOForkSupport && config.DAOForkBlock != nil && config.DAOForkBlock.Cmp(blockNumber) == 0 {
+		misc.ApplyDAOHardFork(statedb)
+	}
+	misc.EnsureCreate2Deployer(config, uint64(block.Time), statedb)
+	var (
+		context vm.BlockContext
+		signer  = types.MakeSigner(config, header.Number, header.Time)
+		err     error
+	)
+	context = core.NewEVMBlockContext(header, chainCtx, nil, config, statedb)
+	vmenv := vm.NewEVM(context, vm.TxContext{}, statedb, config, cfg)
+	if beaconRoot := block.ParentBeaconRoot; beaconRoot != nil {
+		core.ProcessBeaconBlockRoot(*beaconRoot, vmenv, statedb)
+	}
+	if config.IsPrague(blockNumber, uint64(block.Time)) {
+		core.ProcessParentBlockHash(block.ParentHash, vmenv, statedb)
+	}
+	logger.Info("Prepared EVM state")
+	_, _ = fmt.Fprintf(outW, "# Prepared state\n")
+
+	// Iterate over and process the individual transactions
+	for i, tx := range block.Transactions {
+		logger.Info("Processing tx", "i", i, "hash", tx.Hash())
+		_, _ = fmt.Fprintf(outW, "# Processing tx %d\n", i)
```

Comment:
`outW` seems to be a json file writer, json doesn't support comments starting with `#` afaik.

Reply:
It's not a JSON file, it's a list of lines. 1 json trace entry per line. Lines starting with a `#` are comments.